runtime/common/QuditIdTracker.h

```diff
@@ -36,15 +36,15 @@ class QuditIdTracker {
     }
 
     auto next = recycledQudits.front();
-    recycledQudits.pop_front();
+    recycledQudits.erase(recycledQudits.begin());
     return next;
   }
 
   /// @brief Return indices due to qudit deallocation.
   /// If all qudits have been deallocated, reset the
   /// tracker.
   void returnIndex(std::size_t idx) {
-    recycledQudits.push_front(idx);
+    recycledQudits.push_back(idx);
     std::sort(recycledQudits.begin(), recycledQudits.end());
```

Comment:
```suggestion
    std::sort(recycledQudits.begin(), recycledQudits.end(), std::greater<std::size_t>());
```

Reply:
Why is the free list sorted in the first place? Would most recently freed/warmest (or least recently freed/coolest) produce better performance metrics?